ssl/ssl_lib.c

```diff
@@ -4071,6 +4072,7 @@ int SSL_validate_ct(SSL *s)
 
 end:
     CT_POLICY_EVAL_CTX_free(ctx);
+    X509_free(cert);
```

Comment:
Can this just read s->session->peer directly to avoid the locking?

Also, since SSL_validate_ct isn't a public API, it should be lowercased: 'ssl_validate_ct'.